reducers/index.js

```diff
@@ -62,6 +70,17 @@ module.exports = function navigate(state, {type, payload}) {
         stack: stack.splice(index + 1, 1, payload),
       });
 
+    case IMMEDIATELY_RESET_ROUTE_STACK:
+      invariant(
+        payload && payload.length >= 1,
+        'Make sure to pass an array of routes with at least one route.'
+      );
+
+      return Object.assign({}, state, {
+        stack: new Immutable.Stack(payload),
```

Comment:
Actually, doing like you suggested will result in a Stack with the first element the array passed.